facet/tests/derive.rs

```diff
@@ -288,6 +288,13 @@ fn tuple_struct_field_doc_comment() {
     );
 }
 
+#[test]
+fn tuple_struct_generic() {
+    #[derive(Clone, Hash, PartialEq, Eq, ::facet::Facet)]
+    #[repr(transparent)]
+    struct Blah<'a, T: Facet, const C: usize = 3>(T, core::marker::PhantomData<&'a ()>);
```

Comment:
Fwiw I am not sure whether we want to support lifetimes. I am pretty sure `Facet` implementations on anything involving lifetimes are unsound as it likely allows transmuting lifetimes. 

That is I am already worrying about the impls we have for `&str` and `Cow`.

Reply:
> but I imagine we are likely exposing some safe ones that do type checks under the hood at some point and I believe the moment we do we expose lifetime transmutation in safe code.

We certainly need to find a way to do it. I haven't been able to give it much thought yet.

Reply:
Essentially, here's what I'm thinking. `Peek` and `Poke` have exactly one lifetime. They should be able to manipulate anything that has exactly that lifetime, and that's it, in safe code. Ideally, serializers and deserializers should not have to write unsafe code. 

We should deal with a maximum of one lifetime and not be able to change it at all, and do manual checking, I suppose? It might take us a few tries to find something that works, but I don't want to give up that easily.

Reply:
Right, using an invariant lifetime might work out

Reply:
> using an invariant lifetime might work out

Yeah that's my only goal I think.